### PR TITLE
Wrap flect for local use

### PR DESCRIPTION
**Pull request by theunrepentantgeek**
**What this PR does / why we need it**:

We've needed a few custom overrides of `flect` for our context, and we run the risk of having inconsistencies because we're redoing the customization at each location. This PR centralizes our use in one place, a package that can be reused as required.

**How does this PR make you feel**:
gif

**If applicable**:
- [x] this PR contains tests

### CODE REVIEW

File: v2/api/containerinstance/v1beta20211001/container_group_types_gen.go [hunk 1]
```diff
@@ -937,18 +937,18 @@ func (group *ContainerGroup_Spec) AssignProperties_From_ContainerGroup_Spec(sour
 
 	// SubnetIds
 	if source.SubnetIds != nil {
-		subnetIDList := make([]ContainerGroupSubnetId, len(source.SubnetIds))
-		for subnetIDIndex, subnetIDItem := range source.SubnetIds {
+		subnetIdList := make([]ContainerGroupSubnetId, len(source.SubnetIds))
```
Comment (matthchr) [inline v2/api/containerinstance/v1beta20211001/container_group_types_gen.go:940]: minor: previous casing was more correct

File: v2/tools/generator/internal/astmodel/known_locals_set.go [hunk 2]
```diff
@@ -9,9 +9,9 @@ import (
 	"fmt"
 	"strings"
 
-	"github.com/gobuffalo/flect"
-
 	"github.com/Azure/azure-service-operator/v2/internal/set"
+
```
Comment (matthchr) [inline v2/tools/generator/internal/astmodel/known_locals_set.go:13]: minor: I think can remove this space? IIRC all of the `github.com/Azure/azure-service-operator/v2` imports should go together in an ordered block

File: v2/tools/generator/pkg/names/naming.go [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+	{"IP", "IPs"},
+	{"knownAs", "knownAs"},
+	{"KnownAs", "KnownAs"},
+	{"ssh", "sshs"},
+	{"SSH", "SSHs"},
+}
+
+// Singularize returns the singular form of the given name
+func Singularize(name string) string {
+	// work around bug in flect: https://github.com/Azure/azure-service-operator/issues/1454
```
Comment (matthchr) [inline v2/tools/generator/pkg/names/naming.go:35]: minor: Should this be a flect bug link?

File: v2/tools/generator/pkg/names/naming.go [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+
+	singular := flect.Singularize(name)
+
+	// If name starts with an uppercase letter, ensure the result does too
+	return fixCasing(name, singular)
+}
+
+// Pluralize returns the plural form of the given name
+func Pluralize(name string) string {
+	// work around bug in flect: https://github.com/Azure/azure-service-operator/issues/1454
```
Comment (matthchr) [inline v2/tools/generator/pkg/names/naming.go:50]: Same here

File: v2/tools/generator/internal/astmodel/known_locals_set_test.go [hunk 5]
```diff
@@ -99,6 +99,15 @@ func Test_KnownLocalsSet_CreateSingularLocal_ReturnsExpectedResult(t *testing.T)
 	g.Expect(locals.CreateSingularLocal("SubnetIDs")).To(Equal("subnetID"))
 }
 
+func Test_KnownLocalsSet_CreateSingularLocalWithSuffix_ReturnsExpectedResult(t *testing.T) {
+	t.Parallel()
+	g := NewGomegaWithT(t)
+	locals := NewKnownLocalsSet(NewIdentifierFactory())
+
+	g.Expect(locals.CreateSingularLocal("SubnetIds", "Item")).To(Equal("subnetIdItem"))
```
[the remark this answers is not in the document]
Comment (theunrepentantgeek) [reply v2/tools/generator/internal/astmodel/known_locals_set_test.go:107]: As discussed above, the Swagger property is `SubnetIds`

File: v2/tools/generator/pkg/names/naming.go (hunk 3 above)
Comment (theunrepentantgeek) [reply v2/tools/generator/pkg/names/naming.go:35]: That particular bug has been fixed in flect, so I'll remove the link.

File: v2/tools/generator/pkg/names/naming.go (hunk 4 above)
Comment (theunrepentantgeek) [reply v2/tools/generator/pkg/names/naming.go:50]: Removed.

**Review by matthchr — APPROVED** [review]

**Review by super-harsh — APPROVED** [review]

**Pull request merged**